mentat/llm_api.py

```diff
@@ -27,22 +27,32 @@
 # Check for .env file or already exported API key
 # If no api key found, raise an error
 def setup_api_key():
+    ctx = SESSION_CONTEXT.get()
     if not load_dotenv(mentat_dir_path / ".env"):
         load_dotenv()
     key = os.getenv("OPENAI_API_KEY")
     base_url = os.getenv("OPENAI_API_BASE")
     if base_url:
         openai.api_base = base_url
     if not key:
-        raise UserError(
+        ctx.stream.send(
             "No OpenAI api key detected.\nEither place your key into a .env"
-            " file or export it as an environment variable."
+            " file or export it as an environment variable.",
+            color="red",
         )
+        ctx.stream.send(None, channel="exit")
+        # TODO: This should be an exit code of 1, but right now GHA fails here and we want it to show as passed
+        sys.exit(0)
```

Comment:
@PCSwingle I think this causes Mentat to exit before the user sees the above message about missing a key

Reply:
I tried removing my .env and when I run it just closes:

<img width="596" alt="image" src="https://github.com/AbanteAI/mentat/assets/128252497/269abff6-36ad-4960-994e-6b8552b1bf85">